Add Dark Mode

LambArchie opened the pull request:
Resolves #33

Looks like this
![image](https://github.com/user-attachments/assets/948be584-d72a-4820-b7a9-6496178103f2)
![image](https://github.com/user-attachments/assets/6e83262d-5c01-4fb1-a57c-86671d9b1905)
![image](https://github.com/user-attachments/assets/9f44779f-1657-4c3d-96f0-452d958e9fb2)

The approach I took to detecting dark mode (use OS preference) means there is no toggle within the app, but that should be easy to add in a follow-up PR if wanted (and if you want to force dark mode, just add `class="dark"` to `index.html`.

~~Doesn't touch the images within the public folder as while I could edit the background on them, I was hoping someone has the source files and could easily do it instead.~~ Done this now by hand.

~~I've tried to minimise the number of colours defined outside of `colours.ts` so it should be easy to update the colours if wanted.~~ All colours are in `colours.ts` now.

~~The theme isn't properly dynamic, so if you change the OS colour theme preference, the app won't fully update before a refresh. The proper solution to this would require moving the theme into a React Hook and I didn't want to go through that effort due to the large number of places currently importing `colours.ts` but I can do that if wanted.~~ Done this now via CSS variables

Oliveriver commented:
Unfortunately the home page image is a scan of a real drawing, so can't translate to dark mode nicely. I'll have a think about the best way to deal with that.

LambArchie commented:
@Oliveriver I've manually edited the image to remove the white background make it transparent. I did it in [GIMP](https://www.gimp.org/) and I can share the xcf file if wanted. I added feathered edges to make it look nicer and to ensure you can see all the detail, including the black lines.

Oliveriver commented on client/public/landing-image-foreground.png:
Nicely done! This is better for this image all round.

Oliveriver commented on client/src/assets/map/Ank.svg:
```diff
@@ -9,7 +9,7 @@
     <g id="Group" fill-rule="evenodd" stroke-linejoin="round" stroke-width="1.73">
      <path d="M 981.656,258.229 L 974.581,259.999 L 919.749,252.923 L 909.138,258.229 L 903.834,254.694 L 887.912,267.075 L 850.77,258.229 L 831.315,242.309 L 820.7,224.624 L 827.775,217.548 L 827.775,185.713 L 824.239,164.487 L 836.62,164.487 L 866.687,183.942 L 886.143,180.408 L 898.528,183.942 L 939.21,219.317 L 965.741,215.779 L 981.656,219.317 L 985.196,238.773 Z" fill="currentColor" stroke= ...
-     <g id="Group_1" stroke="none" fill="#ffffff">
+     <g class="supply-center-dot" stroke="none" fill="#ffffff">
```
I'm curious why you decided to add a border to these and have them be black to match the region boundaries. White would clash with the order arrows here so I'm not sure I like it.

LambArchie replied:
Adding a border to supply centers was not fully intentional but I kept it as without that the I thought the white arrow would clash.

Given it another think, tried removing the border + changed the order arrows back to black. This has the downside of it's hard to spot the arrow if it's going across unused countries due to the dark grey on black. The arrows over sea are also not the easiest to spot. 
![image](https://github.com/user-attachments/assets/4e9e356c-9234-46d7-8d6b-8d94edc3c609)
Instead I've gone with removing the supply center border but keeping the white arrow. I'm that happy with it but it was the best option I could think of
![image](https://github.com/user-attachments/assets/4d93ebe0-9c4e-45ff-9bba-8eab04f64474)

Oliveriver commented on client/src/components/GameRoot.tsx:
```diff
@@ -17,30 +17,32 @@ const GameRoot = () => {
-        )}
-        <Overlay />
-      </TransformWrapper>
+      <div className="dark:bg-slate-950 dark:text-white">
```
I've tried to keep colour definitions to the `colours` file. Could these be moved there?

LambArchie replied:
Done now

Oliveriver commented on client/src/components/pages/LandingPage.tsx:
```diff
@@ -8,7 +8,7 @@ type LandingPageProps = {
 
 const LandingPage = ({ setViewOption }: LandingPageProps) => (
-  <div className="flex justify-center w-screen min-h-screen">
+  <div className="flex justify-center w-screen min-h-screen dark:text-white">
```
Seeing the same background colours being used for all the roots and pages, I think it would just be neater to change the fragment in `AppRoot` to a `div` and apply the colours there, rather than duplicating everything.

LambArchie replied:
Done now. Also added dynamic dark mode support so the app updates when you toggle your OS colour preference

Oliveriver reviewed (changes requested):
Nice, just a few style comments. I think the colour choice are generally good except for the supply centre dots.

Oliveriver commented:
After using this for a bit, I think the following dark mode colours should be changed:

- The nation/neutral colours in general are too bright. They clash quite a bit with order arrows.
- The button border should be a darker grey, since it's distracting as it is.
- Similarly, the colour for the year divider is too bright. I tried to make it mostly blend into the background to avoid too much clutter in the game world.

LambArchie commented:
@Oliveriver I've updated the colours so they look much nicer now + the rest of your comments.
Also moved styles I had the CSS selector stuff from tailwind to the CSS file to make them easier to read + modify in the future

Oliveriver reviewed (approved):
There are still a few style things I'm uncertain about here, but I'll merge for now and maybe adjust myself at a later date. Thank you for completing this!

Oliveriver commented:
Ah, the CI failed due to linting errors. I'll just override and fix myself to save time.
